dynamic_rest/renderers.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from rest_framework import renderers
+
+
+class DynamicJSONRenderer(renderers.JSONRenderer):
+
+  def render(self, data, accepted_media_type=None, renderer_context=None):
```

Comment:
Hmm, perhaps we can gate the actually sideloading logic with a flag on the viewset for OPTIONS? The hacky yet simplifying part of this solution is that the serializers have to "tag" their data with `_model` and `_pk` such that the renderer can sideload recursively without knowing anything about the request. We probably always want to have the renderer strip these tags out, whether there is any actual sideloading or not.

An interesting case to consider though.. I've actually used sideloading in POSTs and PUTs for the `/applicants/` API in vishnu. For example, when creating an event, I use `POST /events/?sideload[]=users` (old syntax) -- the reason being that the event that comes back has a `creator_id` (user) which in some cases I have never sent to the backend and therefore isnt in the Ember store! So Ember will break unless I sideload the user data along with the POST confirmation response.